### PR TITLE
Minor improvements

### PR DESCRIPTION
**Pull request by TimKoornstra**
# Pull Request: Minor Improvements

This PR introduces a wide range of improvements and optimizations across the codebase and API. The changes aim to enhance performance, readability, and maintainability. 

## General Improvements

- **Code Cleanup:**
  - Removed unused code and functions, streamlining the codebase.

- **Custom Callback Enhancement:**
  - Improved the custom callback with additional logging features and optimized model saving.

- **Logging Overhaul:**
  - Transitioned from `print` statements to `logging`, removing all unnecessary prints. This enhances the clarity and usefulness of log messages.

- **New Optimizer Option - RMSProp:**
  - Introduced RMSProp as an additional optimizer option.

- **Normalization File Handling:**
  - If a normalization file is used during training, it's now saved along with the model.

## Specific Functionality and File Structure Changes

- **CTC Loss Improvement:**
  - Enhanced CTC Loss functionality and relocated it to a different file for better organization.

- **Metrics Modularization:**
  - Moved metrics to a separate file, improving code modularity.

- **CTC Decoding Update:**
  - Adjusted CTC Decoding to normalize using effective timesteps rather than the length of text.

- **Utils to Tokenizer Renaming and Enhancement:**
  - Renamed `Utils` class to `Tokenizer` and improved it with a `__call__` method and more descriptive function names like `encode` and `decode`.

- **Model Loading Function Relocation:**
  - Relocated the `load_model` function for better code organization.

- **Batch Decoding in Tokenizer:**
  - Enhanced the Tokenizer to support batch decoding.

## API-Specific Updates

- **Improved API Logging:**
  - Enhanced logging within the API, including information on the number of images pending preprocessing.

- **API Image Preparator Enhancement:**
  - Made the image preparator in the API more consistent with the training process.
  - This change significantly boosts performance as preprocessing operations can now be executed in parallel.

---

### CODE REVIEW

File: src/visualize/vis_utils.py [hunk 1]
```diff
@@ -11,7 +11,7 @@
 from vis_arg_parser import get_args
```
Comment (Thelukepet) [inline src/visualize/vis_utils.py:11]: imports need to be updated according to latest setup

File: src/visualize/visualize_filters_activations.py [hunk 2]
```diff
@@ -1,6 +1,10 @@
 # Imports
 
 # > Standard Library
+import matplotlib.pyplot as plt
+import tensorflow as tf
+import numpy as np
+from vis_utils import prep_image_for_model, init_pre_trained_model
```
Comment (Thelukepet) [inline src/visualize/visualize_filters_activations.py]: tf, plt and numpy are not in standard library.  Maybe you can fix the other imports here as well? =^)

**Review by Thelukepet — CHANGES_REQUESTED** [review]
Small changes needed for imports
